daemon/cmd/daemon_main.go

```diff
@@ -1918,6 +1919,45 @@ func startDaemon(d *Daemon, restoredEndpoints *endpointRestoreState, cleaner *da
 	if err != nil {
 		log.WithError(err).Error("Unable to store Viper's configuration")
 	}
+
+	checkEndpointBPFProgs(d)
+}
+
+// checkEndpointBPFProgs check to ensure endpoints associated BPF programs remain loaded.
+func checkEndpointBPFProgs(d *Daemon) {
+	const (
+		epsBPFSync         = "eps-bpf-sync"
+		epsBPFSyncDuration = 30 * time.Second
+	)
+	controller.NewManager().UpdateController(
```

Comment:
This is never cleaned up when the daemon stops. This will cause issues in tests that start the agent multiple times.

Reply:
Great point! I'll update...

Reply:
Good point!